tutorials/using-cloud-vpn-with-cisco-asr/index.md

```diff
@@ -0,0 +1,893 @@
+
+    !
+    interface Tunnel1
+     ip address 169.254.0.58 255.255.255.252
+     ip mtu 1400
+     ip tcp adjust-mss 1360
+     tunnel source TenGigabitEthernet0/0/0
+     tunnel mode ipsec ipv4
+     tunnel destination 104.196.200.68
+     tunnel protection ipsec profile VPN_SCALE_TEST_VTI
+    !
+
+#### Configure static or dynamic routing protocol to route traffic into the IPsec tunnel
+
+Statically route traffic toward the network in the GCP to the Tunnel interface.
+
+    ip route 172.16.100.0 255.255.255.0 Tunnel 1
+
+or
+
+BGP is used within the tunnel to exchange prefixes between the GCP and the ASR
+1000 router. The GCP will announce the prefix corresponding to your Cloud.
+
+BGP timers are adjusted to provide more rapid detection of outages.
+
+To advertise additional prefixes to GCP, copy the "network" statement and
+identify the prefix you wish to advertise. Make sure the prefix is present in
+the routing table of the ASR 1000 with a valid next-hop.
+
+    router bgp 65001
+     bgp log-neighbor-changes
+     neighbor 169.254.0.1 description BGP session over Tunnel1
+     neighbor 169.254.0.1 remote-as 65002
+     neighbor 169.254.0.1 timers 20 60 60
+    !
+    address-family ipv4
+     network 10.0.0.0
+     neighbor 169.254.0.1 activate
+    exit-address-family
+
+### Saving the configuration
+
+To save the running configuration and set it as the default startup, run the
+following command on Cisco IOS terminal:
+
+    copy run start
+
+### Test result
+
+    cisco-asr#ping 172.16.100.2 source 10.0.200.1
+    Type escape sequence to abort.
+    Sending 5, 100-byte ICMP Echos to 172.16.100.2, timeout is 2 seconds:
+    Packet sent with a source address of 10.0.200.1
+    !!!!!
+    Success rate is 100 percent (5/5), round-trip min/avg/max = 18/19/20 ms
+
+### Advanced VPN configurations
+
+#### Configure VPN redundancy
+
+![alt_text](https://storage.googleapis.com/gcp-community/tutorials/using-cloud-vpn-with-cisco-asr/GCP-Cisco-ASR-Topology-Redundant.jpg)
+
+Using redundant tunnels ensures continuous availability in the case of a tunnel fails.
+
+If a Cloud VPN tunnel goes down, it restarts automatically. If an entire virtual
+device fails, Cloud VPN automatically instantiates a new one with the same
+configuration, so you don't need to build two Cloud VPN gateways. The new
+gateway and tunnel connect automatically. For hardware appliances such as Cisco
+ASR it is recommended that you deploy atleast 2 ASRs and create VPN tunnels to
+GCP from each for redundancy purposes.
+
+The VPN redundancy configuration example is built based on the IPsec tunnel and
+BGP configuration illustrated above.
+
+##### Cisco ASR
+
+Cisco IOS BGP prefer the path with the highest `LOCAL-PREF`, the BGP routes are
+set with a value of 100 by default, by setting the `LOCAL-PREF` to 200 for the
+routes received from Tunnel1, BGP will choose Tunnel1 as the preferred VPN
+tunnel to the GCP, in the event of Tunnel 1 failure, BGP will reroute the
```

Comment:
is this to GCP? Isn't this setting (on the Cisco router) for ingress traffic from GCP?

Reply:
its for the incoming traffic on Cisco router.